zipp/__init__.py

```diff
@@ -313,6 +321,47 @@ def iterdir(self):
         subs = map(self._next, self.root.namelist())
         return filter(self._is_child, subs)
 
+    def match(self, path_pattern):
+        return pathlib.Path(self.at).match(path_pattern)
+
+    def is_symlink(self):
+        return False  # See #82102
+
+    @contextlib.contextmanager
+    def _scandir(self):  # Needed for glob
+        yield self.iterdir()
+
+    def _make_child_relpath(self, relpath):  # Needed for glob
+        return self / relpath
+
+    def glob(self, pattern):
+        if not pattern:
+            raise ValueError("Unacceptable pattern: {!r}".format(pattern))
+        path = pathlib.Path(self.at)
+        drv, root, pattern_parts = path._flavour.parse_parts((pattern,))
+        if drv or root:
+            raise NotImplementedError("Non-relative patterns are unsupported")
+        if pattern[-1] in (path._flavour.sep, path._flavour.altsep):
+            pattern_parts.append('')
+        selector = pathlib._make_selector(tuple(pattern_parts), path._flavour)
+        for p in selector.select_from(self):
```

Comment:
This operation is failing on Python 3.10 (and probably earlier) with `AttributeError: 'Path' object has no attribute '_accessor'` when the `zipp.Path` isn't instrumented that way. I guess the interfaces were less forgiving in those older versions. I'm not sure what the solution is, but one possibility would be to backport the 'selector' behavior from Python 3.11 or 3.12 and use that.

Reply:
In e587f319826bce03df1dc7233842f054172c1fd2, I addressed the issue by re-writing the glob logic directly.